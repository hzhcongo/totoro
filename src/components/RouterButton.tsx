import React from "react";
import { Button } from "@material-ui/core";
import { useRouter } from "hooks/useRouter";

export interface RouterButtonProps {
  id: string;
  route: string;
  displayText: string;
}

export function RouterButton(props: RouterButtonProps) {
  const router = useRouter();

  return (
    <Button
      id={props.id}
      variant="contained"
      color="primary"
      size="large"
      onClick={() => { router.push(props.route); }}
    >
      {props.displayText}
    </Button>
  );
}
