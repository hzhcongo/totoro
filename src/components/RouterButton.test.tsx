import React from "react";
import renderer, { ReactTestRenderer } from "react-test-renderer";
import {
  createRenderer,
  setupTestEnvironment,
  tearDownTestEnvironment
} from "../utilities/TestUtilities";
import { RouterButton } from "./RouterButton";
import { MemoryRouter } from "react-router-dom";

beforeAll(() => {
	setupTestEnvironment();
})

afterAll(() => {
	tearDownTestEnvironment();
})

describe("Snapshots", () => {
	test("Initial render", () => {
    let testRenderer: ReactTestRenderer = createRenderer(
			<MemoryRouter>
				<RouterButton
					id="testRouterBtnId"
					displayText="testDisplayText"
					route="/testRouter"
				/>
			</MemoryRouter>
		);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});

describe("Functional", () => {
  test("Router button clicked", () => {
    let testRenderer: ReactTestRenderer = createRenderer(
			<MemoryRouter>
				<RouterButton
					id="testRouterBtnId"
					displayText="testDisplayText"
					route="/testRouter"
				/>
			</MemoryRouter>
		);

    const inst = testRenderer.root;
    const button = inst.findByProps({
      id:"testRouterBtnId",
      variant:"contained",
      color:"primary",
      size:"large"
		});
    renderer.act(() => {
      button.props.onClick();
		});
		//TODO: ACTUALLY TEST IF BUTTON ROUTES
  });
});
