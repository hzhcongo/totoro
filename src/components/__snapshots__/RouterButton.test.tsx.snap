// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots Initial render 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
  disabled={false}
  id="testRouterBtnId"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    testDisplayText
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
